modules/base/rendering/renderablemodel.cpp

```diff
@@ -263,13 +267,67 @@ RenderableModel::RenderableModel(const ghoul::Dictionary& dictionary)
     addProperty(_opacity);
     registerUpdateRenderBinFromOpacity();
 
+    if (dictionary.hasKey(KeyForceRenderInvisible)) {
+        _forceRenderInvisible = dictionary.value<bool>(KeyForceRenderInvisible);
 
-    if (dictionary.hasKey(KeyGeometry)) {
-        ghoul::Dictionary dict = dictionary.value<ghoul::Dictionary>(KeyGeometry);
-        for (int i = 1; i <= dict.size(); ++i) {
-            std::string key = std::to_string(i);
-            ghoul::Dictionary geom = dict.value<ghoul::Dictionary>(key);
-            _geometry.push_back(modelgeometry::ModelGeometry::createFromDictionary(geom));
+        if (!_forceRenderInvisible) {
+            // Asset file have specifically said to not render invisible parts,
+            // do not notify in the log if invisible parts are detected and dropped
+            _notifyInvisibleDropped = false;
+        }
+    }
+
+    if (dictionary.hasKey(KeyGeomModelFile)) {
+        std::string file;
+
+        if (dictionary.hasValue<std::string>(KeyGeomModelFile)) {
+            // Handle single file
+            file = absPath(dictionary.value<std::string>(KeyGeomModelFile));
+            _geometry = ghoul::io::ModelReader::ref().loadModel(
+                file,
+                _forceRenderInvisible,
+                _notifyInvisibleDropped
+            );
+        }
+        else if (dictionary.hasValue<ghoul::Dictionary>(KeyGeomModelFile)) {
+            ghoul::Dictionary fileDictionary = dictionary.value<ghoul::Dictionary>(
+                KeyGeomModelFile
+            );
+            std::vector<std::unique_ptr<ghoul::modelgeometry::ModelGeometry>> geometries;
+
+            for (std::string_view k : fileDictionary.keys()) {
+                // Handle each file
+                file = absPath(fileDictionary.value<std::string>(k));
+                geometries.push_back(ghoul::io::ModelReader::ref().loadModel(
+                    file,
+                    _forceRenderInvisible,
+                    _notifyInvisibleDropped
+                ));
+            }
+
+            if (geometries.size() > 0) {
+                ghoul::modelgeometry::ModelGeometry combinedGeometry =
+                std::move(*geometries[0].release());
+
+                 // Combine all models into one ModelGeometry
+                 for (unsigned int i = 1; i < geometries.size(); ++i) {
```

Comment:
```cpp
for (const std::unique_ptr<ghoul::modelgeometry::ModelGeometry>& geometry : geometries) {
    for (const ghoul::io::ModelMesh& mesh : geometry->meshes()) {
        ...
    }
}
```

Reply:
Note that the loop starts at index 1 and not 0 for the geometries. Will change for the others though.

Reply:
Ha!  Yea, you're right  I didn't notice that

Reply:
I noticed now that for both the `ModelMesh` and the `TextureEntry` this will call the copy constructor, which is deleted so I think I’m gonna keep it as it is.

Reply:
Huh.. it shouldn't call the copy constructor if you bind the for loop to a `const ghoul::io::ModelMesh&`;  if it is `const ghoul::io::ModelMesh` then it would, but not with the reference 🤔

Reply:
Well that is what the compiler told me when I tried it. Could it be because the mesh array is accessed with the function `meshes()` and not directly? Because it works fine in all other places.

Reply:
Ah.. yea, it was the second `const` since ofc we can't move out of a `const T&`:
This works:
```cpp
            if (!geometries.empty()) {
                using namespace ghoul::modelgeometry;
                ModelGeometry combinedGeometry = std::move(*geometries[0].release());

                 // Combine all models into one ModelGeometry
                for (const std::unique_ptr<ModelGeometry>& geometry : geometries) {
                    for (ghoul::io::ModelMesh& mesh : geometry->meshes()) {
                        combinedGeometry.meshes().push_back(std::move(mesh));
                    }

                    for (ModelGeometry::TextureEntry& s : geometry->textureStorage()) {
                        combinedGeometry.textureStorage().push_back(std::move(s));
                    }
                }
                _geometry = std::make_unique<ghoul::modelgeometry::ModelGeometry>(
                    std::move(combinedGeometry)
                );
                _geometry->calculateBoundingRadius();
            }
```